ilc/client/setupErrorHandlers.js

```diff
@@ -1,7 +1,89 @@
+import * as singleSpa from 'single-spa';
+import * as uuidv4 from 'uuid/v4';
+import * as ejs from 'ejs/ejs.min.js';
+import apiPage500 from '../api/apiPage500';
+
 const System = window.System;
 const newrelic = window.newrelic;
 
-export default function () {
+// Initializing 500 error page to cache template of this page
+// to avoid a situation when localhost can't return this template in future
+apiPage500.getTemplate()
+    .then(() => console.log('500 error page template cached'))
+    .catch((err) => {
+        noticeError(err, {
+            type: 'FETCH_PAGE_ERROR',
+            name: err.toString(),
+            extraInfo: {
+                errorId: uuidv4(),
+            },
+        });
+    });
+
+const FRAGMENT_KIND = Object.freeze({
+    primary: 'primary',
+    essential: 'essential',
+    regular: 'regular',
+});
+
+const selectFragmentKind = (registryConf, path, appName, slotName) => {
+    const appKind = registryConf.apps[appName].kind;
+    const slotKind = path.slots[slotName] && path.slots[slotName].kind;
+
+    return slotKind || appKind;
+}
+
+const isEssentialOrPrimaryFragment = (fragmentKind) => [
```

Comment:
we use it only one time. probably it's better to remove function and just call `.includes` inside `if`